### PR TITLE
feat(kds): add span for admin requests to zone CPs

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/envoy/admin/kds_client.go [hunk 1]
```diff
@@ -42,124 +53,137 @@ func (k *kdsEnvoyAdminClient) PostQuit(context.Context, *core_mesh.DataplaneReso
...
+	grpc.ReverseUnaryMessage
+	GetError() string
+}
+
+func startTrace(ctx context.Context, tracer trace.Tracer, name string) context.Context {
+	ctx, _ = tracer.Start(
+		ctx,
+		name,
+		trace.WithSpanKind(trace.SpanKindClient),
+		// We fake attributes for the reverse unary gRPC service
```
Comment (slonka) [inline pkg/envoy/admin/kds_client.go:66]: why do we need fake attributes?

**Review by slonka — APPROVED** [review]
just Qs and nitpick

File: pkg/envoy/admin/kds_client.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/envoy/admin/kds_client.go:66]: "Fake" is maybe the wrong word but the problem is that normally these are set by `otelgrpc`, and we have a kind of gRPC service tunneled through around service here. Perhaps I should just remove `RPCSystemGRPC`

File: pkg/envoy/admin/kds_client.go [hunk 2]
```diff
@@ -42,124 +53,137 @@ func (k *kdsEnvoyAdminClient) PostQuit(context.Context, *core_mesh.DataplaneReso
...
+		trace.WithAttributes(
+			semconv.RPCSystemGRPC,
+			semconv.RPCService(reverseUnaryRPCService),
+			semconv.RPCMethod(name),
+		),
+	)
+	return ctx
+}
+
+func doRequest[T message]( // nolint:nonamedreturns
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/envoy/admin/kds_client.go:76]: the `nolint`? So that we can use `defer`.

File: pkg/envoy/admin/kds_client.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/envoy/admin/kds_client.go:66]: Removed

File: pkg/envoy/admin/kds_client.go [hunk 3]
```diff
@@ -42,124 +53,137 @@ func (k *kdsEnvoyAdminClient) PostQuit(context.Context, *core_mesh.DataplaneReso
...
 }
 
-func (k *kdsEnvoyAdminClient) ConfigDump(ctx context.Context, proxy core_model.ResourceWithAddress) ([]byte, error) {
+type message interface {
+	grpc.ReverseUnaryMessage
+	GetError() string
+}
+
+func startTrace(ctx context.Context, tracer trace.Tracer, name string) context.Context {
+	ctx, _ = tracer.Start(
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/envoy/admin/kds_client.go]: changed

**Review by slonka — APPROVED** [review]

**Pull request merged**